File: android-app/app/src/main/java/com/emergance/app/util/RetryPolicy.kt
package com.emergance.app.util

fun nextRetryDelayMs(attempts: Int): Long {
    return when (attempts) {
        0 -> 500
        1 -> 1_000
        2 -> 2_000
        3 -> 4_000
        4 -> 8_000
        5 -> 16_000
        else -> 30_000
    }
}

fun ttlMsByType(type: String): Long {
    return when (type) {
        "SOS_CREATE" -> 86_400_000
        "ASSIGNMENT_OFFER" -> 60_000
        "DRIVER_HEARTBEAT" -> 15_000
        else -> 60_000
    }
}

File: dispatch-desktop/src/core/retryPolicy.ts
const RETRY_SEQUENCE_MS = [500, 1000, 2000, 4000, 8000, 16000];

export function nextRetryDelayMs(attemptsCompleted: number): number {
  if (attemptsCompleted < RETRY_SEQUENCE_MS.length) {
    return RETRY_SEQUENCE_MS[attemptsCompleted];
  }
  return 30_000;
}

export function ttlByMessageType(type: string): number {
  switch (type) {
    case "SOS_CREATE":
      return 86_400_000;
    case "ASSIGNMENT_OFFER":
      return 60_000;
    case "DRIVER_HEARTBEAT":
      return 15_000;
    default:
      return 60_000;
  }
}